.github/workflows/unit-tests.yaml

```diff
@@ -28,9 +28,8 @@ jobs:
         run: |
           python -m pip install --upgrade pip
           pip install -r requirements.txt
```

Comment:
See the comment on the requirements.txt file for context. Lets keep this as is, and just add pytest to requirements-dev **and** add a manual `pip install pytest` here. That way we're not installing different requirements when testing, but we still allow devs to get the benefits of having the listing in the requirements-dev file.